### PR TITLE
[HDInsight] Support kafka rest proxy feature

### PR DESCRIPTION
**Pull request by aim-for-better**
HDInsight
1. support kafka rest proxy feature by add KafkaRestProperties and ClientGroupInfo
2. Add example for KafkaRestProxy

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by fengzhou-msft** [comment]
please fix CI failure for `public.rest-api-specs (ModelValidation) `

**Comment by aim-for-better** [comment]
npm install; npm run prettier-fix

**Comment by aim-for-better** [comment]
> please fix CI failure for `public.rest-api-specs (ModelValidation) `

fixed. But there is a PrettierCheck failed. Could you please help us with this issue?

**Comment by fengzhou-msft** [comment]
> fixed. But there is a PrettierCheck failed. Could you please help us with this issue?

PrettierCheck is not mandatory. But if you would like to fix it, please follow https://aka.ms/AA6h31t.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateHDInsightClusterWithAutoscaleConfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "parameters": {
     "clusterName": "cluster1",
     "resourceGroupName": "rg1",
-    "api-version": "2018-06-01-preview",
+    "api-version": "2015-03-01-preview",
```
Comment (fengzhou-msft) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateHDInsightClusterWithAutoscaleConfig.json]: This change doesn't look right. 
And I think you should put the files for `2018-06-01-preview` under preview folder, not stable.

Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateHDInsightClusterWithAutoscaleConfig.json]: > This change doesn't look right.
> And I think the files for `2018-06-01-preview` should be under preview folder, not stable.

api version is fixed.
And for why the file `2018-06-01-preview` under stable folder, this is a history problem.
This is intentional. It’s a little confusing, but per rule #5 here, whether the swagger spec is in the preview or stable folder “is not a direct analog for whether or not an API Version has the ‘-preview’ suffix or not. SDKs that are generated from 'preview' folder items should indicate to their customers in the most idiomatic way that breaking changes may still be coming.”

Basically the short version is that HDInsight is a GA service that still has an API with a -preview tag. Somebody made the decision several years ago to release stable versions of Hydra/Hyak SDKs for .NET, JavaScript, etc. that consumed this the -preview API. We are now releasing stable swagger-based SDKs so we can at least finally deprecate the Hydra/Hyak SDKs. As a result, we are making a commitment that we will not make breaking changes to this swagger spec version so we can GA some swagger-based SDKs.

Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateHDInsightClusterWithAutoscaleConfig.json]: > This change doesn't look right.
> And I think the files for `2018-06-01-preview` should be under preview folder, not stable.

api version is fixed.
And for why the file `2018-06-01-preview` under stable folder, this is a history problem.

**Review by fengzhou-msft — APPROVED** [review]

**Comment by aim-for-better** [comment]
@fengzhou-msft  When this PR can be merged?

**Comment by aim-for-better** [comment]
Have fixed prettier check

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateKafkaClusterWithKafkaRestProxy.json [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        "clusterDefinition": {
+          "kind": "kafka",
+          "componentVersion": {
+            "Kafka": "2.1"
+          },
+          "configurations": {
+            "gateway": {
+              "restAuthCredential.isEnabled": true,
+              "restAuthCredential.username": "admin",
+              "restAuthCredential.password": "**********"
```
Comment (KrisBash) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateKafkaClusterWithKafkaRestProxy.json]: While this is an example, please ensure that secrets (Password) are never returned in a GET, as suggested here

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json [hunk 3]
```diff
@@ -553,6 +556,28 @@
         }
       }
     },
+    "ClientGroupInfo": {
```
Comment (KrisBash) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json]: It's strongly preferred to add new configurable properties in a new API version. The issue is when a user updates the new properties, and another user does a GET->PUT with an older SDK. The new properties would be omitted from the request body from the downlevel SDK

**Review by KrisBash — APPROVED** [review]
Signing off with comment

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateKafkaClusterWithKafkaRestProxy.json (hunk 2 above)
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateKafkaClusterWithKafkaRestProxy.json]: > While this is an example, please ensure that secrets (Password) are never returned in a GET, as suggested here

Hi @KrisBash  Thanks for your comment. We have user role to control this.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json (hunk 3 above)
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json]: > It's strongly preferred to add new configurable properties in a new API version. The issue is when a user updates the new properties, and another user does a GET->PUT with an older SDK. The new properties would be omitted from the request body from the downlevel SDK

Thanks for your comment. This new properties is not mandatory. And I agree with your opinion.
